### PR TITLE
Add release notes for 7.12.1

### PR DESCRIPTION
**Pull request by AthenaEryma**
This commit adds release notes for 7.12.1.

There were no flagged release highlights for Elasticsearch for this release.

[Preview link](https://elasticsearch_71703.docs-preview.app.elstc.co/guide/en/elasticsearch/reference/7.12/release-notes-7.12.1.html) (once the build finishes)

### CODE REVIEW

**Review by lcawl — APPROVED** [review]
ML content LGTM, thanks!

**Comment by gwbrown** [comment]
@lcawl Thanks for the review and for adding the ML notes. A couple questions (this is the first time I'm doing the release manager thing):
1) Should I have done something differently to include the ML release notes? Or is it expected that those will needed to be added separately?
2) Is this good to merge now, or do we need another review? I ask only because you specified that the ML content looks good.

File: docs/reference/release-notes/7.12.asciidoc [hunk 1]
```diff
@@ -1,3 +1,109 @@
...
+* Fix handling of non-integer port values in `community_id` processor {es-pull}70148[#70148] (issue: {es-issue}70131[#70131])
+* Fix typo in validation for destination port of community ID processor {es-pull}70883[#70883]
+* Templates match indices with date math expressions {es-pull}71433[#71433] (issue: {es-issue}69727[#69727])
+* URI parts processor handles URLs containing spaces {es-pull}71559[#71559] (issue: {es-issue}70947[#70947])
+
+Features/Stats::
+* Avoid return negative value in `CounterMetric` {es-pull}71446[#71446] (issues: {es-issue}52411[#52411], {es-issue}70968[#70968])
+
+Features/Watcher::
+* Enable Setting Master Node Timeout in Watcher Start/Stop Requests {es-pull}70425[#70425]
```
Comment (lcawl) [inline docs/reference/release-notes/7.12.asciidoc:63]:
```suggestion
* Enable setting master node timeout in Watcher start/stop Requests {es-pull}70425[#70425]
```

File: docs/reference/release-notes/7.12.asciidoc [hunk 2]
```diff
@@ -1,3 +1,109 @@
...
+Machine Learning::
+* Make ML memory tracker more robust to flipping on/off master nodes {es-pull}71067[#71067] (issue: {es-issue}68685[#68685])
+
+Mapping::
+* Legacy geo-shape mapper not detecting [points_only] parameter {es-pull}70765[#70765] (issue: {es-issue}70751[#70751])
+
+SQL::
+* Enforce and document dedicated client version compatibility {es-pull}70451[#70451] (issue: {es-issue}70400[#70400])
+* Fix manifest version tag in Tableau connector {es-pull}71524[#71524]
+* Resolve attributes recursively for improved subquery support {es-pull}69765[#69765] (issues: {es-issue}2[#2], {es-issue}4[#4], {es-issue}6[#6], {es-issue}7[#7], {es-issue}8[#8], {es-issue}22[#22], {es-issue}67237[#67237])
```
Comment (lcawl) [inline docs/reference/release-notes/7.12.asciidoc:86]: I think the script was erroneously pulling these extra issues from the code example:
```suggestion
* Resolve attributes recursively for improved subquery support {es-pull}69765[#69765] (issue:  {es-issue}67237[#67237])
```

File: docs/reference/release-notes/7.12.asciidoc [hunk 3]
```diff
@@ -1,3 +1,109 @@
...
+
+Search::
+* Fix exception when merging completion suggestions {es-pull}70414[#70414] (issue: {es-issue}70328[#70328])
+* Fix query cache reporting negative used memory {es-pull}70273[#70273] (issue: {es-issue}55434[#55434])
+* Fix search states of CCS requests in mixed cluster {es-pull}70948[#70948] (issue: {es-issue}52741[#52741])
+
+Snapshot/Restore::
+* Avoid atomic overwrite tests on FS repositories {es-pull}70483[#70483] (issue: {es-issue}70303[#70303])
+* Drop alloc filters on mount of searchable snapshot {es-pull}70007[#70007] (issue: {es-issue}69759[#69759])
+* Fix Source Only Snapshot Permanently Broken on Broken `_snapshot` Directory {es-pull}71459[#71459]
```
Comment (lcawl) [inline docs/reference/release-notes/7.12.asciidoc:97]:
```suggestion
* Fix source only snapshot permanently broken on broken `_snapshot` directory {es-pull}71459[#71459]
```

File: docs/reference/release-notes/7.12.asciidoc [hunk 4]
```diff
@@ -1,3 +1,109 @@
...
+* Fix `isDedicatedVotingOnlyNode` {es-pull}71358[#71358] (issue: {es-issue}71352[#71352])
+* Fix human readable xcontent for snapshots in progress and deletion {es-pull}70256[#70256]
+* Fix wrongly computed offset in checksum {es-pull}69441[#69441] (issues: {es-issue}69415[#69415], {es-issue}69437[#69437])
+* Only allocate partial shards to nodes with cache {es-pull}69904[#69904]
+* Optimized read footer checksum from `FileInfo` {es-pull}69415[#69415]
+* Reduce Memory Use of Parallel Azure Blob Deletes {es-pull}71330[#71330] (issue: {es-issue}71267[#71267])
```
Comment (lcawl) [inline docs/reference/release-notes/7.12.asciidoc:103]:
```suggestion
* Reduce memory use of parallel Azure blob deletes {es-pull}71330[#71330] (issue: {es-issue}71267[#71267])
```

File: docs/reference/release-notes/7.12.asciidoc [hunk 5]
```diff
@@ -1,3 +1,109 @@
...
+* Stop Blocking Snapshot Deletes Due to Concurrency Limits {es-pull}71050[#71050]
```
Comment (lcawl) [inline docs/reference/release-notes/7.12.asciidoc:104]:
```suggestion
* Stop blocking snapshot deletes due to concurrency limits {es-pull}71050[#71050]
```

**Comment by lcawl** [comment]
> 1. Should I have done something differently to include the ML release notes? Or is it expected that those will needed to be added separately?

Unfortunately, this is always a manual process. The ML writers do it, so it's not something the release coordinator has to worry about. 

> 2. Is this good to merge now, or do we need another review? I ask only because you specified that the ML content looks good.

Typically an Elasticsearch writer reviews as well, but I think most of them are away so I did a quick check of the rest of the PRs too and submitted some more comments.

**Comment by gwbrown** [comment]
@lcawl Thanks both the answers and the review are much appreciated!

**Pull request merged**